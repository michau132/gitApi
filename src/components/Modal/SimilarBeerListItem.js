import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import Image from '../Image';

const SimilarBeerListStyle = styled.li`
  width: 180px;
  padding: 10px;
  background-color: #fff;
  display: inline-block;
  transition: .5s all;
  margin-bottom: 10px;
  box-shadow: 0px 0px 4px 1px rgba(122,122,122,0.9);
  
  ${({ theme }) => theme.media.tablet} {
    margin-right: 15px;
    box-shadow: 0px 0px 4px 1px rgba(122,122,122,0.9);
    
    :hover {
      transform: scale(1.05);
      box-shadow: 0px 0px 19px 4px rgba(122,122,122,1);
    }
  }
`;


const SimilarBeerListImgWrapper = styled.div`
  height: 100px;
`;

const BeerName = styled.div`
  font-weight: bold;
`;


const SimilarBeerListItem = ({ id, image_url, name }) => (
  <SimilarBeerListStyle key={id}>
    <Link to={{ pathname: `/trainingGit/beer/${id}`, state: { modal: true } }}>
      <SimilarBeerListImgWrapper>
        <Image src={image_url} alt={name} />
      </SimilarBeerListImgWrapper>
      <BeerName>{name}</BeerName>
    </Link>
  </SimilarBeerListStyle>
);

SimilarBeerListItem.propTypes = {
  id: PropTypes.number.isRequired,
  image_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default SimilarBeerListItem;
