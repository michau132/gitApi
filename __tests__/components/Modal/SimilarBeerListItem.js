import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SimilarBeerListItem from '../../../src/components/Modal/SimilarBeerListItem';

describe('<SimilarBeerListItem /> component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      id: 1,
      image_url: 'https://via.placeholder.com/150',
      name: 'Beer',
    };
    wrapper = shallow(<SimilarBeerListItem {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders Image correctly', () => {
    const img = wrapper.find('Image');
    expect(img.prop('src')).toEqual(props.image_url);
    expect(img.prop('alt')).toEqual(props.name);
  });
});
