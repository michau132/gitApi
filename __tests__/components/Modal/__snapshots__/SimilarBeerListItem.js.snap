// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimilarBeerListItem /> component matching snapshot 1`] = `
<styled.li
  key="1"
>
  <Link
    replace={false}
    to={
      Object {
        "pathname": "/beer/1",
        "state": Object {
          "modal": true,
        },
      }
    }
  >
    <styled.div>
      <Image
        alt="Beer"
        src="https://via.placeholder.com/150"
      />
    </styled.div>
    <styled.div>
      Beer
    </styled.div>
  </Link>
</styled.li>
`;
